### PR TITLE
Adds surface

### PR DESCRIPTION
**Pull request by jlaura**
This PR adds a surface class, originally implemented in AutoCNet by @jessemapel. I have moved this class into knoten as this seems to be a more natural home. Having a surface is necessary for sensor model operations with the CSM. This PR adds the surface classes, tests, updated `csm` to use the surface, and updates the CSM tests to use the surface models.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [x] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

**Comment by jlaura** [comment]
The Py3.8 failure is a libpoppler issue related to GDAL/ALE. Is 3.8 expected to be working with GDAL/ALE in the same env?

File: knoten/csm.py [hunk 1]
```diff
@@ -127,7 +127,7 @@ def generate_ground_point(dem, image_pt, camera):
           GeoDataset object generated from Plio off of a Digital Elevation
           Model (DEM)
     image_pt : tuple
-               Pair of x, y coordinates in pixel space
+               Pair of x, y (line, sample) coordinates in pixel space
```
Comment (Kelvinrr) [inline knoten/csm.py]: wouldn't x be sample and y be line?

File: requirements_test.txt [hunk 2]
```diff
@@ -0,0 +1,2 @@
+pytest
+pytest-xdist
```
Comment (Kelvinrr) [inline requirements_test.txt]: we usually just leave the test requirements in the env.yml and leave them out of the meta.yml. Wouldn't this require an extra install if you wanted to run tests? As it seems it needed to be added to the build script.

Comment (jlaura) [reply requirements_test.txt]: Sounds good I'll move them back in and will check the meta.yml for dependencies.

**Review by Kelvinrr — APPROVED** [review]

**Pull request merged**